### ISSUE TITLE
Migrate away from modal context

### ISSUE THREAD

**Issue by joseplayero** [reorproject/reor#439]
We should have a shadcn dialog component that we use for every modal and then whenever a modal is needed, the component that generates the component can just import the child content that goes inside the modal/dialog. That will make the codebase simpler and easier to understand in my opinion

@milaiwi do you have thoughts on matter?

**Comment by milaiwi** [comment]
Yeah, that'd be much better. So essentially defining an `export ReusableDialog` that another component imports and passes it's own `CustomContent`.
